sdk/rust/nym-sdk/src/mixnet/client.rs

```diff
@@ -415,193 +399,80 @@ where
-    /// client identity, the client encryption key, and the gateway identity.
-    pub fn nym_address(&self) -> &Recipient {
-        &self.nym_address
-    }
-
-    /// Get a shallow clone of [`MixnetClientSender`]. Useful if you want split the send and
-    /// receive logic in different locations.
-    pub fn sender(&self) -> MixnetClientSender {
-        MixnetClientSender {
-            client_input: self.client_input.clone(),
-        }
-    }
-
-    /// Get a shallow clone of [`ConnectionCommandSender`]. This is useful if you want to e.g
-    /// explicitly close a transmission lane that is still sending data even though it should
-    /// cancel.
-    pub fn connection_command_sender(&self) -> ConnectionCommandSender {
-        self.client_input.connection_command_sender.clone()
-    }
-
-    /// Get a shallow clone of [`LaneQueueLengths`]. This is useful to manually implement some form
-    /// of backpressure logic.
-    pub fn shared_lane_queue_lengths(&self) -> LaneQueueLengths {
-        self.client_state.shared_lane_queue_lengths.clone()
-    }
-
-    /// Change the network topology used by this client for constructing sphinx packets into the
-    /// provided one.
-    pub async fn manually_overwrite_topology(&self, new_topology: NymTopology) {
-        self.client_state
-            .topology_accessor
-            .manually_change_topology(new_topology)
-            .await
-    }
+    pub async fn connect_to_mixnet_via_socks5(self) -> Result<Socks5MixnetClient>
+    where
+        <B as ReplyStorageBackend>::StorageError: Sync + Send,
+    {
+        let key_manager = self.key_manager.clone();
+        let socks5_config = self.socks5_config.clone().ok_or(Error::NoSocks5ConfigSet)?;
+        let (mut started_client, nym_address) = self.connect_to_mixnet_common().await?;
 
-    /// Gets the value of the currently used network topology.
-    pub async fn read_current_topology(&self) -> Option<NymTopology> {
-        self.client_state.topology_accessor.current_topology().await
-    }
+        let client_input = started_client.client_input.register_producer();
+        let client_output = started_client.client_output.register_consumer();
+        let client_state = started_client.client_state;
 
-    /// Restore default topology refreshing behaviour of this client.
-    pub fn restore_automatic_topology_refreshing(&self) {
-        self.client_state.topology_accessor.release_manual_control()
+        let socks5_listener_boot_time_secs = 5;
+        nym_socks5_client_core::NymClient::start_socks5_listener(
+            &socks5_config,
+            client_input,
+            client_output,
+            client_state.clone(),
+            nym_address,
+            started_client.task_manager.subscribe(),
+        );
+        log::debug!(
+            "Waiting for {} seconds for the socks5 channel to boot",
+            socks5_listener_boot_time_secs
+        );
+        std::thread::sleep(std::time::Duration::from_secs(
```

Comment:
No need to sleep, the socks5 client should send a status message once the connection is established. I think nym-connect listens to this to know when to show connected in the UI